job/job.go

```diff
@@ -3,7 +3,9 @@ package job
 import (
 	"container/ring"
 	gocontext "context"
+	"crypto/rand"
```

Comment:
We don't want to use crypto/rand which can block if there is not enough entropy - use math/rand